packages/web/pages/assets/[denom].tsx

```diff
@@ -77,7 +77,19 @@ const AssetInfoView: FunctionComponent<AssetInfoPageStaticProps> = observer(
     const [previousTrade, setPreviousTrade] =
       useLocalStorage<PreviousTrade>(SwapPreviousTradeKey);
 
-    const { title, details, coinGeckoId, asset: asset } = useAssetInfo();
+    const {
+      title,
+      details,
+      coinGeckoId,
+      asset,
+      denom: routerDenom,
+    } = useAssetInfo();
+
+    useEffect(() => {
+      if (routerDenom !== asset?.coinMinimalDenom) {
```

Comment:
Per coderrabbit I agree we should check for null to ensure the check is what we expect